src/Compilers/Core/Portable/MetadataReference/CompilationReference.cs

```diff
@@ -110,5 +111,37 @@ public override string Display
                 return Compilation.AssemblyName;
             }
         }
+
+        public bool Equals(CompilationReference other)
+        {
+            if (object.ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            if (other == null)
+            {
+                return false;
+            }
+
+            // MetadataProperty implements value equality
+            return object.ReferenceEquals(this.Compilation, other.Compilation) && object.Equals(this.Properties, other.Properties);
```

Comment:
Agreed. Also do two lines, since then it makes it nice to see the parallel-ness of this.